Neos.Fusion/Classes/Core/Runtime.php

```diff
@@ -568,6 +614,7 @@ protected function prepareContextForFusionObject(AbstractFusionObject $fusionObj
         }
 
         if (isset($newContextArray)) {
+            $this->fusionGlobals->requireContextToNotInterfereWithGlobals($newContextArray);
             $this->pushContextArray($newContextArray);
```

Comment:
Maybe change signature of pushContextArray to
```
function pushContextArray(array $newContextArray, bool $preventOverwritingGlobals = false): void
```
that way it would be backwards compatible and still confine the checks to the pushContextArray method.

Reply:
hmm sorry i think this flag is to complex. Instead i could imagine a separate `pushContextAndOverrideGlobal` method - or keep everything as is on good trust basis ^^

Reply:
That would be fine for me aswell but the logic imho belongs into the runtime. Fusion.Form would have to get a flow version detection for that but this is ok.

Reply:
Fusion.Form could imo as well use `method_exists($runtime, 'pushContextAndOverrideGlobals')` and based upon that call either that or fallback to `pushContext`.

And then yes i can move the `requireContextToNotInterfereWithGlobals` logic into `pushContext` ^^.

Additionally i would deprecate `pushContextAndOverrideGlobals` (without a removal) as its highly discouraged.

Reply:
Add a note to the deprecation that this is currently in use by fusion form. Maybe annotation as internal would make sense. Not much opinion there from my side.

Reply:
Implemented :D